torchgeo/datasets/cropharvest.py

```diff
@@ -0,0 +1,673 @@
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            DatasetNotFoundError: if ``download=False`` but
+                dataset is missing or checksum fails
+        """
+        self.root = root
+        self.transforms = transforms
+        self.checksum = checksum
+        self.download = download
+
+        self._verify()
+
+        self.files = self._load_features(self.root)
+        self.labels = self._load_labels(self.root)
+
+    def __getitem__(self, index: int) -> dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            single pixel timeseries array and label at that index
+        """
+        files = self.files[index]
+        data = self._load_array(files["chip"])
+
+        label = self._load_label(files["index"], files["dataset"])
+        sample = {"array": data, "label": label}
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.files)
+
+    def _load_features(self, root: str) -> list[dict[str, str]]:
+        """Return the paths of the files in the dataset.
+
+        Args:
+            root: root dir of dataset
+
+        Returns:
+            list of dicts containing path for each of hd5 single pixel time series and
+            its key for associated data
+        """
+        files = []
+        chips = glob.glob(
+            os.path.join(root, self.file_dict["features"]["extracted_filename"], "*.h5")
+        )
+        chips = sorted(os.path.basename(chip) for chip in chips)
+        for chip in chips:
+            chip_path = os.path.join(
+                root, self.file_dict["features"]["extracted_filename"], chip
+            )
+            index = chip.split("_")[0]
+            dataset = chip.split("_")[1][:-3]
+            files.append(dict(chip=chip_path, index=index, dataset=dataset))
+        return files
+
+    def _load_labels(self, root: str) -> pd.DataFrame:
+        """Return the paths of the files in the dataset.
+
+        Args:
+            root: root dir of dataset
+
+        Returns:
+            pandas dataframe containing label data for each feature
+        """
+        filename = self.file_dict["labels"]["extracted_filename"]
+        with open(os.path.join(root, filename), encoding="utf8") as f:
+            data = json.load(f)
+
+            pd.json_normalize(data["features"])
+            df = pd.json_normalize(data["features"])
```

Comment:
Why are we running this twice? What does `json_normalize` do that we can't do with `json.load`?